src/Renderer/LineMaterial.js

```diff
@@ -0,0 +1,44 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+
+import * as THREE from 'three';
+import BasicMaterial from './BasicMaterial';
+import Fetcher from '../Core/Commander/Providers/Fetcher';
+import LineVS from './Shader/LineVS.glsl';
+import LineFS from './Shader/LineFS.glsl';
+
+const LineMaterial = function LineMaterial(options) {
+    BasicMaterial.call(this);
+
+    if (options === undefined) {
+        throw new Error('options is required');
+    }
+
+    this.fragmentShader = this.fragmentShaderHeader + LineFS;
+    this.vertexShader = this.vertexShaderHeader + LineVS;
+
+    this.wireframe = false;
+
+    const texture = options.texture ? Fetcher.texture.load(options.texture) : undefined;
+
+    this.uniforms.THICKNESS = { value: options.linewidth };
+    this.uniforms.MITER_LIMIT = { value: true };
+    this.uniforms.WIN_SCALE = { value: new THREE.Vector2(window.innerWidth, window.innerHeight) };
```

Comment:
Actually simply doing something similar to:
```
this.uniforms.WIN_SCALE = { value: c3DEngine().viewerDiv.size() }
```
should be enough (ie: pointing the uniforms to an Vector2 object stored elsewhere and kept up to date)